nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,232 @@
+"""
+region_extractor
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+
+from skimage.feature import peak_local_max
+from skimage.segmentation import random_walker
+
+from nilearn import plotting
+from nilearn._utils import check_niimg
+from nilearn._utils.extmath import fast_abs_percentile
+from nilearn.image import iter_img, new_img_like
+from nilearn.image.image import _smooth_array
+from nilearn._utils.niimg_conversions import concat_niimgs
+from nilearn._utils.compat import _basestring
+
+
+def extract_seperate_regions(img, img_data, min_size):
```

Comment:
```img_data``` is the thresholded image taken as input in this function. I am calling ```img``` since I am using it to take affine and create a ```new_img_like```.